monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -551,6 +550,7 @@ def get_report(cls):
             raise RuntimeError("Agent repository does not exists")
 
         if cls.report_is_outdated():
-            cls._report = safe_generate_regular_report()
+            with cls._report_generation_lock:
+                cls._report = cls.generate_report()
```

Comment:
```suggestion
            ```suggestion
            if not cls._generating_report.is_set():
                cls._generating_report.set()
                cls._report = cls.generate_report()
                cls.generating_report.clear()
```
```